FinniversKit/Sources/Base Components/ButtonStyle.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  Copyright © 2020 FINN AS. All rights reserved.
+//
+
+import SwiftUI
+
+@available(iOS 13.0, *)
+public struct InlineFlatStyle: ButtonStyle {
+    let font: Font
+    let textColor: Color
+
+    public init(size: Button.Size = .normal, textColor: Color = .btnPrimary) {
+        self.font = size == .normal ? .finnFont(.bodyStrong) : .finnFont(.detailStrong)
+        self.textColor = textColor
+    }
+
+    public func makeBody(configuration: Configuration) -> some View {
+        configuration
+            .label
+            .font(font)
+            .foregroundColor(textColor)
+    }
+}
+
+@available(iOS 13.0, *)
+struct FlatStyle: ButtonStyle {
+    let font: Font
+    let textColor: Color
+
+    init(size: Button.Size = .normal, textColor: Color = .btnPrimary) {
+        self.font = size == .normal ? .finnFont(.bodyStrong) : .finnFont(.detailStrong)
+        self.textColor = textColor
+    }
+
+    func makeBody(configuration: Configuration) -> some View {
+        HStack {
+            Spacer()
+            configuration
+                .label
+                .font(font)
+                .foregroundColor(textColor)
+            Spacer()
+        }
+        .padding(.vertical, .spacingS)
+        .padding(.horizontal, .spacingM)
+    }
+}
+
+@available(iOS 13.0, *)
+struct DefaultStyle: ButtonStyle {
+    let font: Font
+    let textColor: Color
+
+    init(size: Button.Size = .normal, textColor: Color = .btnPrimary) {
+        self.font = size == .normal ? .finnFont(.bodyStrong) : .finnFont(.detailStrong)
+        self.textColor = textColor
+    }
+
+    func makeBody(configuration: Configuration) -> some View {
+        HStack {
+            Spacer()
+            configuration.label
+                .font(font)
+                .foregroundColor(textColor)
+            Spacer()
+        }
+        .padding(.vertical, .spacingS)
+        .padding(.horizontal, .spacingM)
+        .background(Color.bgPrimary)
```

Comment:
It seems like everything up until setting the background is the same as in `FlatStyle`. What do you think about doing something like:
```swift
FlatStyle(size: size, textColor: textColor)
   .makeBody(configuration: configuration)
   .background(Color.bgPrimary)
   .roundedBorder(radius: .spacingS, color: .btnDisabled)
```

Or perhaps there's even better way of reusing this code.

Reply:
I wrote the code initially and now I was looking for ways of reusing. Nice suggestion!